spec/realtime/codecs.test.js

```diff
@@ -0,0 +1,65 @@
+"use strict";
+
+define(['shared_helper', 'ably-delta-codec'], function(helper, AblyDeltaCodec) {
+	var exports = {},
+		displayError = helper.displayError,
+		closeAndFinish = helper.closeAndFinish,
+		monitorConnection = helper.monitorConnection;
+
+	var equals = function(a, b) {
+		return JSON.stringify(a) === JSON.stringify(b);
+	};
+
+	exports.setupMessage = function(test) {
+		test.expect(1);
+		helper.setupApp(function(err) {
+			if(err) {
+				test.ok(false, displayError(err));
+			} else {
+				test.ok(true, 'setup app');
+			}
+			test.done();
+		});
+	};
+
+	exports.deltaCodec = function(test) {
+		test.expect(3);
+		var testName = 'delta-codec';
+		try {
+			var testData = [
+				{ foo: 'bar', count: 1, status: 'active' },
+				{ foo: 'bar', count: 2, status: 'active' },
+				{ foo: 'bar', count: 2, status: 'inactive' }
+			];
+
+			var realtime = helper.AblyRealtime({
+				codecs: {
+					'vcdiff': new AblyDeltaCodec.AblyVcdiffCodec()
+				}
+			});
+			var channel = realtime.channels.get('[?delta=vcdiff]' + testName);
+
+			channel.subscribe(function(message) {
+				var index = Number(message.name);
+				test.ok(equals(testData[index], message.data), 'Check message.data');
```

Comment:
Am I missing something, how does this test that the delta is working?  If `var channel = realtime.channels.get('[?delta=vcdiff]' + testName);` was changed to `var channel = realtime.channels.get(testName);` wouldn't this test still pass?  If so, that's a flawed test as it's not testing that deltas are being applied.  I would argue we should have another test as well to make sure providing the `codecs` in the constructor `ClientOptions`, without passing `[?delta=vcdiff]` has no side-effect.

Reply:
<quote>
Am I missing something, how does this test that the delta is working? If var channel = realtime.channels.get('[?delta=vcdiff]' + testName); was changed to var channel = realtime.channels.get(testName); wouldn't this test still pass? If so, that's a flawed test as it's not testing that deltas are being applied.
</quote>

You are not missing anything. This is actually fixed here - [LINK](https://github.com/ably/ably-js/pull/615/files#diff-2bc3dbc1297758ec55d556915690486dR60)

<quote>
I would argue we should have another test as well to make sure providing the codecs in the constructor ClientOptions, without passing [?delta=vcdiff] has no side-effect.
</quote>

This is actually a very good idea. It is coming.

Reply:
Test added in https://github.com/ably/ably-js/pull/615/commits/cece42863925fab6a92967a6a75755ced1d73d97